CMakeLists.txt

```diff
@@ -240,6 +242,9 @@ if(CLR_CMAKE_PLATFORM_XPLAT)
         # -Wno-return-type # switch unreachable code
         # -Wno-switch # switch values not handled
 
+    add_compile_options(-fexceptions)
+    add_compile_options(-fstack-protector-all)
```

Comment:
> What's the perf impact with -fstack-protector-all?

Comparing to before ? Almost, none. PAL was enabling that and we depend on PAL. In general ? i.e. can we disable it to get more perf ? We `maybe` could (since this is the version with -all flag, it will be adding on everywhere) but this is already being done in an optimized way. 

What we already have + [add more space for stack][1/2] + check/compare[1] + cleanup[3] + stack_guard?

My guess, up to 5/8 additional instructions per call (worst case), 5/8 additional inst. vs an average CC function, it is less than 1 percent ?

My earlier PR,  I had enabled only `-fstack-protector` for similar reasons, but IMO after your comment to that PR `all` makes more sense for CC. I am not sure if we have lots of critical functions under the threshold of `should we include this too`?